test/extended/util/cvo.go

```diff
@@ -0,0 +1,57 @@
+package util
+
+import (
+	"fmt"
+	"time"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	configv1 "github.com/openshift/api/config/v1"
+	configclientv1 "github.com/openshift/client-go/config/clientset/versioned/typed/config/v1"
+)
+
+// WaitForClusterProgression waits for a cluster-level configuration change to propagate across all operators.
+// This polls all ClusterVersion objects and waits until the following states are stable:
+//
+// 1. Available = true
+// 2. Progressing = false
+// 3. Failing = false
+//
+// If a ClusterVersion reports it is failing, this will abort with an error.
+func WaitForClusterProgression(cv configclientv1.ClusterVersionsGetter) error {
+	return wait.Poll(6*time.Second, 10*time.Minute, func() (bool, error) {
+		clusterVersions, err := cv.ClusterVersions().List(metav1.ListOptions{})
+		if err != nil {
+			// Wait if underlying service is unavailable - indicates apiserver churn
+			if errors.IsServiceUnavailable(err) {
```

Comment:
the tests currently time out because the API server is always unavailable

Reply:
why would the api server never be available?